File: src/Application Specific/Classes/Entity Collections/PartsToBeCredited/PartsToBeCredited.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace PartsToBeCrediteds

        Public Class PartsToBeCredited

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "PartsToBeCredited Properties"


            Private _PartsToBeCreditedID As Integer = 0
            Public ReadOnly Property PartsToBeCreditedID() As Integer
                Get
                    Return _PartsToBeCreditedID
                End Get
            End Property
            Public WriteOnly Property SetPartsToBeCreditedID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartsToBeCreditedID", Value)
                End Set
            End Property


            Private _OrderID As Integer = 0
            Public ReadOnly Property OrderID() As Integer
                Get
                    Return _OrderID
                End Get
            End Property
            Public WriteOnly Property SetOrderID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderID", Value)
                End Set
            End Property


            Private _SupplierID As Integer = 0
            Public ReadOnly Property SupplierID() As Integer
                Get
                    Return _SupplierID
                End Get
            End Property
            Public WriteOnly Property SetSupplierID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SupplierID", Value)
                End Set
            End Property


            Private _PartID As Integer = 0
            Public ReadOnly Property PartID() As Integer
                Get
                    Return _PartID
                End Get
            End Property
            Public WriteOnly Property SetPartID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartID", Value)
                End Set
            End Property

            Private _Qty As Integer = 0
            Public ReadOnly Property Qty() As Integer
                Get
                    Return _Qty
                End Get
            End Property
            Public WriteOnly Property SetQty() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Qty", Value)
                End Set
            End Property
             

            Private _ManuallyAdded As Boolean = False
            Public ReadOnly Property ManuallyAdded() As Integer
                Get
                    Return _ManuallyAdded
                End Get
            End Property
            Public WriteOnly Property SetManuallyAdded() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ManuallyAdded", Value)
                End Set
            End Property


            Private _StatusID As Integer = 0
            Public ReadOnly Property StatusID() As Integer
                Get
                    Return _StatusID
                End Get
            End Property
            Public WriteOnly Property SetStatusID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_StatusID", Value)
                End Set
            End Property


            Private _CreditReceived As Double = 0
            Public ReadOnly Property CreditReceived() As Double
                Get
                    Return _CreditReceived
                End Get
            End Property
            Public WriteOnly Property SetCreditReceived() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_CreditReceived", Value)
                End Set
            End Property


            Private _OrderReference As String = 0
            Public ReadOnly Property OrderReference() As String
                Get
                    Return _OrderReference
                End Get
            End Property
            Public WriteOnly Property SetOrderReference() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderReference", Value)
                End Set
            End Property

            Private _PartOrderID As Integer = 0
            Public ReadOnly Property PartOrderID() As Integer
                Get
                    Return _PartOrderID
                End Get
            End Property
            Public WriteOnly Property SetPartOrderID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartOrderID", Value)
                End Set
            End Property


            Private _ReturnRef As String = 0
            Public ReadOnly Property ReturnRef() As String
                Get
                    Return _ReturnRef
                End Get
            End Property
            Public WriteOnly Property SetReturnRef() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ReturnRef", Value)
                End Set
            End Property


#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/PartsToBeCredited/PartsToBeCredited.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace PartsToBeCrediteds
    {
        public class PartsToBeCredited
        {
            private DataTypeValidator _dataTypeValidator;

            public PartsToBeCredited()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _PartsToBeCreditedID = 0;

            public int PartsToBeCreditedID
            {
                get
                {
                    return _PartsToBeCreditedID;
                }
            }

            public object SetPartsToBeCreditedID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartsToBeCreditedID", value);
                }
            }

            private int _OrderID = 0;

            public int OrderID
            {
                get
                {
                    return _OrderID;
                }
            }

            public object SetOrderID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderID", value);
                }
            }

            private int _SupplierID = 0;

            public int SupplierID
            {
                get
                {
                    return _SupplierID;
                }
            }

            public object SetSupplierID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SupplierID", value);
                }
            }

            private int _PartID = 0;

            public int PartID
            {
                get
                {
                    return _PartID;
                }
            }

            public object SetPartID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartID", value);
                }
            }

            private int _Qty = 0;

            public int Qty
            {
                get
                {
                    return _Qty;
                }
            }

            public object SetQty
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Qty", value);
                }
            }

            private bool _ManuallyAdded = false;

            public int ManuallyAdded
            {
                get
                {
                    return Conversions.ToInteger(_ManuallyAdded);
                }
            }

            public object SetManuallyAdded
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ManuallyAdded", value);
                }
            }

            private int _StatusID = 0;

            public int StatusID
            {
                get
                {
                    return _StatusID;
                }
            }

            public object SetStatusID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_StatusID", value);
                }
            }

            private double _CreditReceived = 0;

            public double CreditReceived
            {
                get
                {
                    return _CreditReceived;
                }
            }

            public object SetCreditReceived
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_CreditReceived", value);
                }
            }

            private string _OrderReference = 0.ToString();

            public string OrderReference
            {
                get
                {
                    return _OrderReference;
                }
            }

            public object SetOrderReference
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderReference", value);
                }
            }

            private int _PartOrderID = 0;

            public int PartOrderID
            {
                get
                {
                    return _PartOrderID;
                }
            }

            public object SetPartOrderID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartOrderID", value);
                }
            }

            private string _ReturnRef = 0.ToString();

            public string ReturnRef
            {
                get
                {
                    return _ReturnRef;
                }
            }

            public object SetReturnRef
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ReturnRef", value);
                }
            }


            
        }
    }
}